### ISSUE TITLE
DisplaySettings: Changing resolution causes crash

### ISSUE THREAD

**Issue by auronandace** [SerenityOS/serenity#7245]
The default resolution of 1024x768 works fine. When I try to change the resolution to 1600x900 then WindowServer seems to crash and displays the following text squished at the top (which I am unable to read):

![WindowServer_Crash](https://user-images.githubusercontent.com/81362994/118633775-bb15a700-b7c9-11eb-9bc8-5af2285e87b1.png)

I am then forced to use Qemu's menu to forcibly shut Serenity (version 1.0.g3ee092c) down. Here is the relevant text from the console:

`DisplaySettings(31:31): main window: 0x03d10c08`
`[#0 WindowServer(22:22)]: BochsGraphicsAdapter: resolution set to 1600x900`
`[WindowServer(22:22)]: ASSERTION FAILED: !m_userspace_framebuffer_region`
`[WindowServer(22:22)]: ../../Kernel/Graphics/FramebufferDevice.cpp:35 in virtual Kernel::KResultOr<Kernel::Region*> Kernel::FramebufferDevice::mmap(Kernel::Process&, Kernel::FileDescription&, const Kernel::Range&, u64, int, bool)`
`[#0 WindowServer(22:22)]: 0xc05bd4bc  abort +0x4b`
`[#0 WindowServer(22:22)]: 0xc05b07cd  debug_handler +0x0`
`[#0 WindowServer(22:22)]: 0xc018f5ce  Kernel::FramebufferDevice::mmap(Kernel::Process&, Kernel::FileDescription&, Kernel::Range const&, unsigned long long, int, bool) +0x708`
`[#0 WindowServer(22:22)]: 0xc024e4c9  Kernel::FileDescription::mmap(Kernel::Process&, Kernel::Range const&, unsigned long long, int, bool) +0x271`
`[#0 WindowServer(22:22)]: 0xc04517c8  Kernel::Process::sys$mmap(AK::Userspace<Kernel::Syscall::SC_mmap_params const*>) +0x2bb8`
`[#0 WindowServer(22:22)]: 0xc03dea5c  syscall_handler +0x1e7d`
`[#0 WindowServer(22:22)]: 0xc03dcba7  syscall_asm_entry +0x31`

**Comment by rwl4** [comment]
I'm seeing the exact same thing. My host system is Pop! OS.

**Comment by auronandace** [comment]
Looking at Kernel/Graphics/FramebufferDevice.cpp:35 I find this:

`VERIFY(!m_userspace_framebuffer_region);`

Just above it is a FIXME that says:

```
// FIXME: We rely on the fact that only the WindowServer will mmap the framebuffer
// and only once when starting to work with it. If other program wants to do so, we need to fix this.
```

Also on lines 111-114:

```
bool FramebufferDevice::set_resolution(size_t, size_t, size_t)
        {
            VERIFY_NOT_REACHED();
        }
```

Does the above mean that screen resolution can currently only be set at boot via a config file and that the resolution handling hasn't been implemented yet in the new Graphics system?

**Comment by auronandace** [comment]
Fixed by #7258
